### PR TITLE
add supplemental constructor for StaticReserve

### PR DESCRIPTION
**Pull request by jd-lara**

### CODE REVIEW

File: src/models/supplemental_constructors.jl [hunk 1]
```diff
@@ -55,3 +55,11 @@ function Bus(number, name, bustype::String, angle, voltage, voltagelimits, basev
     return Bus(number, name, get_enum_value(BusType, bustype), angle, voltage,
                voltagelimits, basevoltage, InfrastructureSystemsInternal())
 end
+
+"""Allows to contruct a reserve from and iterator """
```
Comment (daniel-thom) [inline src/models/supplemental_constructors.jl]:
```suggestion
"""Allows construction of a reserve from an iterator."""
```

File: src/models/supplemental_constructors.jl [hunk 2]
```diff
@@ -55,3 +55,11 @@ function Bus(number, name, bustype::String, angle, voltage, voltagelimits, basev
...
+function StaticReserve(name, contributingdevices::IS.FlattenIteratorWrapper,
+                       timeframe, requirement, _forecasts, InfrastructureSystemsInternal)
+
```
Comment (daniel-thom) [inline src/models/supplemental_constructors.jl]: Why do you always make the first line of a function a blank line?  Is that in some style guide?

Comment (jd-lara) [reply src/models/supplemental_constructors.jl]: I just find it easier to read, no specific reason

Comment (daniel-thom) [reply src/models/supplemental_constructors.jl]: Ok, no problem.  Just no that you are the only person who does it, so we can always tell who wrote a function.  That isn't necessarily a good thing.  It's better if all code in a package looks the same.

**Pull request merged**